### ISSUE TITLE
[Windows10] Freeze when click the opening chat window button.

### ISSUE THREAD

**Issue by nassau-tk** [hrydgard/ppsspp#14044]
### What happens?
Freeze when click the opening chat window button.
But, This issue has a strange point.

The point is "It will happen after using [Alt + Enter] a few times".
Yes, It is "Short cut of Change Full Screen & Returen to window mode" on Windows.
If use the short cut by 2-3 times and click opening chat button then PPSSPP freeze.

If change to full screen & windows mode by double click on screen then it will not happen freeze.

### What *should* happen?
um....Should Permit the shortcut? :sweat:

### Please test newest version in https://buildbot.orphis.net/ppsspp/

PPSSPP Ver.：1.10.3-1714
OS：Windows10（64-bit）Japanese
CPU：Ryzen7 3800X（8-core）
GPU：RADEON RADEON RX5700XT
Resolution：2560×1440
RAM：32GB

**Comment by ANR2ME** [comment]
I can't seems to reproduce this on Metal Slug XX with OpenGL backend.

1). Is this issue happened on all games?
2). Is this issue happened on all backend?
3). Do you need to play multiplayer first? (since Chat button is always there since the game booted)
4). Do you need to be in fullscreen  when clicking the Chat button to trigger the freezes? or windowed mode will also freezes?

**Comment by nassau-tk** [comment]
Ans. for 1) I tested MH2G /Metal Slug XX / Danball Senki.
Ans. for 2) I tested Vulkan/OpenGL.
Ans. for 3) Sorry,Did you said meaning "Do you want play on multiplayer mode?" don't you? um... I'm not need multiplayer mode now. I tested multiplayer mode for my future.😓
Ans. for 4) Full screen / Window mode both.

Maybe....How to way freeze by step below.
1. Click chat button.
2. Push Alt+Enter few times.
3. Click chat button.
4. Push Alt+Enter few times.

https://user-images.githubusercontent.com/48179091/106421039-4da26c80-649f-11eb-93f9-f25a495e4259.mp4

https://user-images.githubusercontent.com/48179091/106421049-56933e00-649f-11eb-8df8-8ba13aeb5e3b.mp4

**Comment by nassau-tk** [comment]
Of course, If turn off chat button on PPSSPP system then the freeze is not happen.

**Comment by ANR2ME** [comment]
Okay i found a vulkan issue after i clicked Chat button and immediately press Alt-Enter to go to fullscreen without waiting for the chat windows to be shown completely.
![image](https://user-images.githubusercontent.com/7974720/106423607-568d4100-6493-11eb-8be5-2f173cbd32a9.png)
![image](https://user-images.githubusercontent.com/7974720/106423643-6ad13e00-6493-11eb-8d0e-6cd399165f27.png)
![image](https://user-images.githubusercontent.com/7974720/106423778-acfa7f80-6493-11eb-9689-096e3e477626.png)
```
45:06:553 CRI ADX Audi E[G3D]: Vulkan\VulkanDebug.cpp:86 VKDEBUG: ERROR(validation:719760766) Validation Error: [ VUID-vkCmdSetScissor-offset-00597 ] Object 0: handle = 0x246cfa2af28, type = VK_OBJECT_TYPE_COMMAND_BUFFER; | MessageID = 0x2ae6ad7e | vkCmdSetScissor: offset.y + extent.height (=495 + 4294967073 = 4294967568) of pScissors[0] will overflow int32_t. The Vulkan spec states: Evaluation of (offset.y + extent.height) must not cause a signed integer addition overflow for any element of pScissors (https://vulkan.lunarg.com/doc/view/1.2.162.1/windows/1.2-extensions/vkspec.html#VUID-vkCmdSetScissor-offset-00597)
```
But i can still continue the game, PPSSPP didn't freeze forever or crashed.

**Comment by ANR2ME** [comment]
Btw @nassau-tk is that permanent freeze only happened while the game is playing a video?

**Comment by nassau-tk** [comment]
No, Of course It was happen same issue in playable game screen.

**Comment by ANR2ME** [comment]
Hmm.. that is strange, i can't reproduce these freeze... may be because i'm using Geforce?

**Comment by nassau-tk** [comment]
OMG!
RADEON issue coming Again for me!? 🤣

**Comment by ANR2ME** [comment]
LOL may be try opening the Debug's Log console to see if there is any error message at the time the freeze occurred?

**Comment by nassau-tk** [comment]
Please tell me about log.
Is this setting is OK?
![image](https://user-images.githubusercontent.com/48179091/106425718-a8d85d00-64a7-11eb-90b2-ef16dbd56c38.png)

**Comment by ANR2ME** [comment]
yes, info logs should be okay to see the Log console live

**Comment by nassau-tk** [comment]
Please, Check the log!

https://gist.github.com/nassau-tk/7f2f6d98bbdf1471f76d73ec2e7eb837

**Comment by ANR2ME** [comment]
Hmm.. doesn't seems to have any Error, only a few Warnings
May be need to wait the devs or anyone with visual studio that can reproduce it (may be someone with Radeon LOL)

**Comment by nassau-tk** [comment]
I tested on software rendering too.
Result is same.

Curse of RADEON...?:scream:

**Comment by nassau-tk** [comment]
Uuuuuuum.....

I tested on other PC "Intel & GeForce".
Result : Not Freeze !

Ryzen & RADEON has a many problem...:sweat:

**Comment by hrydgard** [comment]
Sounds like an issue with negative window sizes or something in the chat window, causing negative scissor rectangles.

Should be fairly easily fixable, I'll take a look tonight.

**Comment by ANR2ME** [comment]
The freezes he experiencing happened on both vulkan and opengl, so may be it's a different issue.

**Comment by nassau-tk** [comment]
@ANR2ME 

Yes, My Radeon is...below.
PowerColor Red Devil Radeon RX 5700 XT.
This RADEON has "Normal & O/C" 2-mode.
But, O/C is "Default".:grin:

**Comment by ANR2ME** [comment]
May be try it without overclocking it?

**Comment by nassau-tk** [comment]
Good morning!

I tried to change O/C mode toSilent mode on RADEON.
![20210203_072415](https://user-images.githubusercontent.com/48179091/106672782-ffeb4880-65f3-11eb-80bc-4516eabcaee2.jpg)

Result : Not change.:sweat: It's freeze.:snowman:

**Comment by nassau-tk** [comment]
@hrydgard 
@ANR2ME 

I think if this issue will happen at rare case of specification then it should be re-mark milestone to v1.12.0.
I think  that It should not be pending to release to v1.11.0 by this issue.
How do you think about it?:confused:

 rare case of specification = some RADEON? = minority? :grin:

**Comment by nassau-tk** [comment]
@hrydgard 

If you got a time then please re-mark milestone!
Thanks!

**Comment by nassau-tk** [comment]
@ANR2ME 

Ah....Please tell me.
If click chat button on bottom of left then chat dialog appear on upper of left?
Is this normal-process?
![image](https://user-images.githubusercontent.com/48179091/108167337-dde7df00-7138-11eb-960e-0ec913cdccac.png)

By the way, This issue still happen "My Windows-PC" on OpenGl & Vulkan both.

**Comment by ANR2ME** [comment]
It never happened to me, i'm not familiar with the chat feature, it was taken directly from https://github.com/hrydgard/ppsspp/pull/9090

**Comment by unknownbrackets** [comment]
Most likely that has to do with the setting to use the native Windows keyboard.

-[Unknown]

**Comment by nassau-tk** [comment]
Probably, If click the chat button at window mode then appear the small dialog window.
Exept full screen mode.

**Comment by ANR2ME** [comment]
You need to disable "Use system native keyboard" if you don't want to have that popup window
![image](https://user-images.githubusercontent.com/7974720/108218973-d4c23680-7167-11eb-8adc-3dcd0a3b84cf.png)

Btw, i tried to enable it and when i clicked the "Chat" button right after launching a game (the ISO is still being loaded) PPSSPP became stop responding LOL

**Comment by nassau-tk** [comment]
@ANR2ME 

Yes, It was cause!!
You found it!!
Great!!:wink:

If turn on the "Use system native keybord" and use "Alt + Enter (shortcut of changing Fullscreen)" and click "Chat button" then freeze happen.
Moreover, The PPSSPP system is setting "turned on" on default isn't it?

Is it bug? or not?:confused:

**Comment by nassau-tk** [comment]
By the way....

If turn off it.
We can't use system of "native text input "(  #12732  ).
That system needs a "Use system native keybord".:confused:

**Comment by hrydgard** [comment]
I would call any hang a bug - we're gonna need to find a way to prevent the hang from happening.

**Comment by unknownbrackets** [comment]
Well, making the input dialog async fixes that, which is probably the better thing to do anyway.

-[Unknown]

**Comment by nassau-tk** [comment]
@unknownbrackets 

um...

My Main PC is became like a below.(Ryzen+RADEON Win10)

![chat](https://user-images.githubusercontent.com/48179091/108461168-8a52ce00-72bd-11eb-859a-24760422fbe9.png)

But, My sub PC is not appear the small dialog window.(Intel+NVIDIA Win10)

Is this has a possibility of setting of OS?

**Comment by nassau-tk** [comment]
but, Freeze is dissappear!!

**Comment by unknownbrackets** [comment]
Well, that's what the setting does - I'm not really sure if it makes sense for chat or not, but at least it doesn't freeze now.

I kinda think we should change that to be some sort of inline dialog overlay, rather than a modal pop up.

-[Unknown]

**Comment by unknownbrackets** [comment]
I think the freeze is gone - is this issue fixed?  Or is some part of the problem still there?

-[Unknown]

**Comment by nassau-tk** [comment]
u----m.....

1. Freeze issue was "solved".

2. Text input dialogue working difference on Fullscreen or Windowmode. 

And probably this issue has rerlation for "input of2byte characters".
1-byte input user has not relation.

**On Window mode**
[System native KB ==>On]
Click Chat ==> Appear the Text input dialogue on top of left.
[System native KB ==>Off]
Click Chat ==> Not appear the Text input dialogue on top of left and when input 2byte characters then appear the characters top of left by IME system directly.

**On Fullscreen mode**
[System native KB ==>On] & [System native KB ==>Off]
Click Chat ==> Not appear the Text input dialogue on top of left and when input 2byte characters then appear the characters top of left by IME system directly.

I think that It should be appear the any text on the "Chat text space".

And why mouse pointer is disappearing on the game screen? (Appear on tool bar.)
When changed it?
It's not comfortable.

**Comment by ANR2ME** [comment]
> And why mouse pointer is disappearing on the game screen? (Appear on tool bar.)
Probably because PSP game doesn't use pointer, thus unnecessary to show it on game screen (the same reason why mouse pointer are hidden when watching movie on media players)

**Comment by nassau-tk** [comment]
Yes,So l think about lt should be same as some media player.
